src/ray/stats/metric_exporter_client_test.cc

```diff
@@ -0,0 +1,184 @@
+    for (auto &point : points) {
+      if (point.metric_name.find(".min") != std::string::npos) {
+        lastest_hist_min = point.value;
+      }
+      if (point.metric_name.find(".mean") != std::string::npos) {
+        lastest_hist_mean = point.value;
+      }
+      if (point.metric_name.find(".max") != std::string::npos) {
+        lastest_hist_max = point.value;
+      }
+    }
+  }
+
+ private:
+  static int client1_count;
+  static int client1_value;
+  static double lastest_hist_min;
+  static double lastest_hist_mean;
+  static double lastest_hist_max;
+};
+
+class MockExporterClient2 : public MetricExporterDecorator {
+ public:
+  MockExporterClient2(std::shared_ptr<MetricExporterClient> exporter)
+      : MetricExporterDecorator(exporter) {
+    client2_count = 0;
+  }
+  void ReportMetrics(const MetricPoints &points) override {
+    MetricExporterDecorator::ReportMetrics(points);
+    client2_count += points.size();
+    RAY_LOG(DEBUG) << "Client 2 " << client2_count << " last metric "
+                   << points.back().metric_name << ", value " << points.back().value;
+    client2_value = points.back().value;
+  }
+  static int GetCount() { return client2_count; }
+  static void ResetCount() { client2_count = 0; }
+  static int GetValue() { return client2_value; }
+
+ private:
+  static int client2_count;
+  static int client2_value;
+};
+
+class MetricExporterClientTest : public ::testing::Test {
+ public:
+  void SetUp() {
+    const stats::TagsType global_tags = {{stats::LanguageKey, "CPP"},
+                                         {stats::WorkerPidKey, "1000"}};
+    ray::stats::Init("127.0.0.1:8888", global_tags, false);
+    std::shared_ptr<MetricExporterClient> exporter(new stats::StdoutExporterClient());
+    std::shared_ptr<MetricExporterClient> mock1(new MockExporterClient1(exporter));
+    std::shared_ptr<MetricExporterClient> mock2(new MockExporterClient2(mock1));
+    MetricExporter::Register(mock2, kMockReportBatchSize);
+  }
+
+  void Shutdown() {
+    MockExporterClient1::ResetCount();
+    MockExporterClient2::ResetCount();
+  }
+};
+
+int MockExporterClient1::client1_count;
+double MockExporterClient1::lastest_hist_min;
+double MockExporterClient1::lastest_hist_mean;
+double MockExporterClient1::lastest_hist_max;
+int MockExporterClient2::client2_count;
+int MockExporterClient1::client1_value;
+int MockExporterClient2::client2_value;
+
+/// Default report flush interval is 10s, so we may wait a while for data
+/// exporting.
+uint32_t kReportFlushInterval = 10000;
+
+bool DoubleEqualTo(double value, double compared_value) {
+  return value >= compared_value - 1e-5 && value <= compared_value + 1e-5;
+}
+
+TEST_F(MetricExporterClientTest, decorator_test) {
+  // Export client should emit at least once in 10 seconds.
+  for (size_t i = 0; i < 100; ++i) {
+    stats::CurrentWorker().Record(i + 1);
+  }
+  std::this_thread::sleep_for(std::chrono::milliseconds(kReportFlushInterval + 200));
```

Comment:
Does this mean the test will always run for 10s? Can we mock out the timer instead of relying on this sleep?

Reply:
At the very least we should reduce this period so it doesn't take so long

Reply:
> At the very least we should reduce this period so it doesn't take so long

We can't do it in this opencensus release version since it's hard code in opencensus.

Reply:
> At the very least we should reduce this period so it doesn't take so long

But stats exporter interval can be modified in https://github.com/census-instrumentation/opencensus-cpp/archive/b14a5c0dcc2da8a7fc438fab637845c73438b703.zip, 
```
        absl::Duration report_interval = absl::Seconds(5);
        opencensus::stats::StatsExporter::SetInterval(report_interval);
```
Actually, I'll import this commit in next PR.